### PR TITLE
Feature: eng 465 update all the codegens

### PR DESCRIPTION
**Pull request by oddgrd**
## Description of change

Rewrite the codegen of shuttle_main to use shuttle_runtime to start the users project binary. Also refactored the `local_run` function to work with this change. Refactoring deployer remains, but I think that should be done in a follow-up PR.

TODO:

- [x] Clean up the codegen, I essentially just unwrapped the wrapper currently. 
- [x] Refactor unwrap of return type.
- [x] Fix the tests.

## How Has This Been Tested (if applicable)?

Local runs of two axum examples, one hello_world and one with the static files resource.

### CODE REVIEW

File: runtime/src/legacy/mod.rs [hunk 1]
```diff
@@ -100,21 +100,29 @@ where
 {
     type Service: Service;
 
-    async fn load(self, factory: Fac) -> Result<Self::Service, shuttle_service::Error>;
+    async fn load(
+        self,
+        factory: Fac,
+        logger: shuttle_service::Logger,
```
Comment (chesedo) [inline runtime/src/legacy/mod.rs:106]: Can we move `Logger` down to the runtime?

File: runtime/src/legacy/mod.rs [hunk 2]
```diff
@@ -154,9 +162,12 @@ where
         );
         trace!("got factory");
 
+        let logs_tx = self.logs_tx.clone();
+        let logger = shuttle_service::Logger::new(logs_tx, deployment_id);
```
Comment (chesedo) [inline runtime/src/legacy/mod.rs:166]: Thanks for the hookup O!

File: service/src/lib.rs [hunk 3]
```diff
@@ -227,6 +227,7 @@ mod logger;
 pub use logger::Logger;
 
 pub use shuttle_common::database;
+pub use shuttle_common::storage_manager::StorageManager;
```
Comment (chesedo) [inline service/src/lib.rs:230]: Can we move this down to runtime too?

**Review by chesedo — APPROVED** [review]
Looking good :partying_face: LGTM

File: runtime/src/legacy/mod.rs (hunk 1 above)
Comment (oddgrd) [reply runtime/src/legacy/mod.rs:106]: Sure!

File: service/src/lib.rs (hunk 3 above)
Comment (oddgrd) [reply service/src/lib.rs:227]: I assume you mean move the re-export, which yes, we can :smile: The storage manager itself is used in a lot of crates that I'm not sure yet if will depend on shuttle-runtime. Maybe we can revisit this when we know.

**Pull request merged**